src/lambda/handler-runner/go-runner/GoRunner.js

```diff
@@ -0,0 +1,186 @@
+      this.writeText = v3Utils.writeText
+      this.v3Utils = v3Utils
+    }
+
+    // Make sure we have the mock-lambda runner
+    spawnSync('go', ['get', 'github.com/icarus-sullivan/mock-lambda'])
+  }
+
+  cleanup() {
+    // Clean up session timeout
+    if (this.#handle !== null) {
+      clearTimeout(this.#handle)
+    }
+
+    try {
+      rmdirSync(this.#tmpPath, { recursive: true })
+    } catch (e) {
+      // @ignore
+    }
+
+    this.#tmpFile = null
+    this.#tmpPath = null
+  }
+
+  _parsePayload(value) {
+    const log = []
+    let payload
+
+    for (const item of value.split(EOL)) {
+      if (item.indexOf(PAYLOAD_IDENTIFIER) === -1) {
+        log.push(item)
+      } else if (item.indexOf(PAYLOAD_IDENTIFIER) !== -1) {
+        try {
+          const {
+            offline_payload: { success, error },
+          } = parse(item)
+          if (success) {
+            payload = success
+          } else if (error) {
+            payload = error
+          }
+        } catch (err) {
+          // @ignore
+        }
+      }
+    }
+
+    // Log to console in case engineers want to see the rest of the info
+    if (this.log) {
+      this.log(log.join(EOL))
+    } else {
+      console.log(log.join(EOL))
+    }
+
+    return payload
+  }
+
+  clearCredentials() {
+    this.#credentials = null
+    this.#handle = null
+  }
+
+  async run(event, context) {
+    const { dir } = pathParse(this.#handlerPath)
+    const handlerCodeRoot = dir.split(delimiter).slice(0, -1).join(delimiter)
+    const handlerCode = readFileSync(`${this.#handlerPath}.go`, 'utf8')
+    this.#tmpPath = resolve(handlerCodeRoot, 'tmp')
+    this.#tmpFile = resolve(this.#tmpPath, 'main.go')
+
+    const out = handlerCode.replace(
+      '"github.com/aws/aws-lambda-go/lambda"',
+      'lambda "github.com/icarus-sullivan/mock-lambda"',
+    )
+
+    try {
+      mkdirSync(this.#tmpPath, { recursive: true })
+    } catch (e) {
+      // @ignore
+    }
+
+    try {
+      writeFileSync(this.#tmpFile, out, 'utf8')
+    } catch (e) {
+      // @ignore
+    }
+
+    // Remove our root, since we want to invoke go relatively
+    const cwdPath = this.#tmpFile.replace(`${cwd()}${delimiter}`, '')
+
+    // Get go env to run this locally
+    if (!this.#goEnv) {
+      const goEnvResponse = spawnSync('go', ['env'], {
+        encoding: 'utf-8',
+      })
+
+      const goEnvString = goEnvResponse.stdout || goEnvResponse.stderr
+      this.#goEnv = goEnvString.split(EOL).reduce((a, b) => {
+        const [k, v] = b.split('="')
+        // eslint-disable-next-line no-param-reassign
+        a[k] = v ? v.slice(0, -1) : ''
+        return a
+      }, {})
+    }
+
+    // Get session credentials to pass to go
+    if (!this.#credentials) {
+      const sts = new AWS.STS()
+
+      const { Credentials } = await sts
+        .getSessionToken({
+          DurationSeconds: MIN_ALLOWED_SESSION_DURATION_S, // 900-inf
+        })
+        .promise()
```

Comment:
Uncertain about this, but credentials are not passed into go without setting them in env.